src/time_travel/patchers/time_patcher.py

```diff
@@ -1,27 +1,34 @@
 """A patch to the time module."""
 
-from .basic_patcher import BasicPatcher
+from .base_patcher import BasePatcher
 
 import mock
+import time
 
 
-class TimePatcher(BasicPatcher):
+class TimePatcher(BasePatcher):
     """Patcher of the time module.
     
     Patching:
         - time
         - sleep
     """
     
-    def __init__(self, *args, **kwargs):
+    def __init__(self, **kwargs):
         """Create the patch."""
-        super(TimePatcher, self).__init__(*args, **kwargs)
+        super(TimePatcher, self).__init__(patcher_module=__name__, **kwargs)
         
-        self.patches.append(
-            mock.patch('time.time', side_effect=self._get_timestamp))
+    def get_patched_module(self):
+        """Do more stuff."""
+        return time
         
-        self.patches.append(
-            mock.patch('time.sleep', side_effect=self._advance_time_stamp))
+    def get_patch_actions(self):
+        """Return generator containing all patches to do."""
+        return [
+            ('time', time.time, mock.Mock(side_effect=self._get_timestamp)),
```

Comment:
You don't need to use Mock here at all. You need to override the original functions with the `side effects` functions themselves.

Reply:
Removed!